src/codegen/encoding/mod.rs

```diff
@@ -614,35 +1047,42 @@ fn finish_array_loop(for_loop: &ForLoop, vartab: &mut Vartable, cfg: &mut Contro
 fn load_struct_member(ty: Type, expr: Expression, field: usize) -> Expression {
     if ty.is_fixed_reference_type() {
         // We should not dereference a struct or fixed array
-        return Expression::StructMember(Loc::Codegen, ty, Box::new(expr), field);
+        return Expression::StructMember(Codegen, ty, expr.into(), field);
     }
+    let s = Expression::StructMember(Codegen, Type::Ref(ty.clone().into()), expr.into(), field);
+    Expression::Load(Codegen, ty, s.into())
+}
+
+/// Get the array length inside a variable.
+fn array_length(arr: &Expression, vartab: &mut Vartable, cfg: &mut ControlFlowGraph) -> Expression {
```

Comment:
This function should be called `array_outer_length`, as it cannot be used for any dimension. In addition, the doc comment should say so.